pkg/repository/knowledgebase.go

```diff
@@ -0,0 +1,139 @@
+	DeleteKnowledgeBase(ctx context.Context, kb KnowledgeBase) error
+}
+
+type KnowledgeBase struct {
+	ID          uint       `gorm:"column:id;primaryKey" json:"id"`
+	KbID        string     `gorm:"column:kb_id;size:255;not null;unique" json:"kb_id"`
+	Name        string     `gorm:"column:name;size:255;not null" json:"name"`
+	Description string     `gorm:"column:description;size:1023" json:"description"`
+	Tags        TagsArray  `gorm:"column:tags;type:VARCHAR(255)[]" json:"tags"`
+	Owner       string     `gorm:"column:owner;size:255;not null" json:"owner"`
+	CreateTime  time.Time  `gorm:"column:create_time;not null;default:CURRENT_TIMESTAMP" json:"create_time"`
+	UpdateTime  time.Time  `gorm:"column:update_time;not null;default:CURRENT_TIMESTAMP" json:"update_time"`
+	DeleteTime  *time.Time `gorm:"column:delete_time" json:"delete_time"`
+}
+
+// table columns map
+type KnowledgeBaseColumns struct {
+	ID          string
+	KbID        string
+	Name        string
+	Description string
+	Tags        string
+	Owner       string
+	CreateTime  string
+	UpdateTime  string
+	DeleteTime  string
+}
+
+var KnowledgeBaseColumn = KnowledgeBaseColumns{
+	ID:          "id",
+	KbID:        "kb_id",
+	Name:        "name",
+	Description: "description",
+	Tags:        "tags",
+	Owner:       "owner",
+	CreateTime:  "create_time",
+	UpdateTime:  "update_time",
+	DeleteTime:  "delete_time",
+}
+
+type TagsArray []string
+
+// CreateKnowledgeBase inserts a new KnowledgeBase record into the database.
+func (r *Repository) CreateKnowledgeBase(ctx context.Context, kb KnowledgeBase) (*KnowledgeBase, error) {
+	// Set the creation and update times to the current time if they are not already set
+	if kb.CreateTime.IsZero() {
+		kb.CreateTime = time.Now().UTC()
+	}
+	if kb.UpdateTime.IsZero() {
+		kb.UpdateTime = time.Now().UTC()
+	}
+
+	if err := r.db.WithContext(ctx).Create(&kb).Error; err != nil {
+		return nil, err
+	}
+
+	return &kb, nil
+}
+
+// GetKnowledgeBase fetches all KnowledgeBase records from the database, excluding soft-deleted ones.
+func (r *Repository) GetKnowledgeBase(ctx context.Context) ([]KnowledgeBase, error) {
+	var knowledgeBases []KnowledgeBase
+
+	// Exclude records where DeleteTime is not null
+	if err := r.db.WithContext(ctx).Where("delete_time IS NULL").Find(&knowledgeBases).Error; err != nil {
+		return nil, err
+	}
+
+	return knowledgeBases, nil
+}
+
+// UpdateKnowledgeBase updates a KnowledgeBase record in the database except for CreateTime and DeleteTime.
+func (r *Repository) UpdateKnowledgeBase(ctx context.Context, kb KnowledgeBase) (*KnowledgeBase, error) {
+	// Fetch the existing record to ensure it exists and to get the CreateTime and DeleteTime fields
+	var existingKB KnowledgeBase
+	if err := r.db.WithContext(ctx).First(&existingKB, kb.ID).Error; err != nil {
+		if err == gorm.ErrRecordNotFound {
+			return nil, err
+		}
+		return nil, err
+	}
+
+	// Preserve CreateTime and DeleteTime
```

Comment:
I remember we don't need to handle the timestamp update logic. Gorm can take care of it. If you have time, you can trace the current mgmt-backend codebase to see how it works.

Reply:
fixed. using gorm's autoUpdateTime tag